### PR TITLE
[16.10] Various bits of hardening from #4604

### PR DESCRIPTION
**Pull request by natefoo**
Because these merges were so complicated and I made some mistakes with #4801, I'm going to PR each forward merge and ensure tests pass.

This also includes a backport of testing fixes from 17.01 to get the tests to actually run on 16.10.

### CODE REVIEW

**Comment by natefoo** [comment]
This is missing changes, do not merge.

**Comment by natefoo** [comment]
The missing changes were the ones not present in 16.07 but present in newer versions.

File: tox.ini [hunk 1]
```diff
@@ -43,14 +44,14 @@ whitelist_externals = bash
...
-    flake8-import-order==0.9
+    flake8-import-order>=0.9
 
 [testenv:py27-lint-imports-include-list]
 commands = bash .ci/flake8_wrapper_imports.sh
 whitelist_externals = bash
 skip_install = True
 deps =
-    flake8==3.0.4
+    flake8==3.2.1
```
Comment (nsoranzo) [inline tox.ini:54]: I think this is causing the failure of the corresponding Travis job with lots of errors that I fixed in 17.01, revert?

Comment (natefoo) [reply tox.ini:54]: The tests won't run at all with older versions, how about we add `I201` to the ignore list for 16.10?

File: lib/galaxy/tools/deps/container_resolvers/mulled.py [hunk 2]
```diff
@@ -33,9 +33,9 @@
 
 
 def list_cached_mulled_images(namespace=None):
-    command = build_docker_images_command(truncate=True, sudo_docker=False)
-    command = "%s | tail -n +2 | tr -s ' ' | cut -d' ' -f1,2" % command
-    images_and_versions = check_output(command)
+    command = build_docker_images_command(truncate=True, sudo=False)
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/container_resolvers/mulled.py:36]: This is also backporting commit 9f4add70d0967ccb6ddf1bd2d1393685ef639e97 from release 17.01, which fixed a nasty bug 👍

**Comment by nsoranzo** [comment]
In #4801 we missed the backport of `s/yaml.load/yaml.safe_load/` for `scripts/grt.py` (which was later renamed `scripts/grt/export.py`). I think that re-adding it here should be enough.

**Review by nsoranzo — APPROVED** [review]

**Comment by natefoo** [comment]
Good catch! I made a list of all modified files in each release to compare for 17.01 through 17.09. Hopefully the next one will go smoother. I really appreciate your help!

**Comment by jmchilton** [comment]
+1

**Pull request merged**